polars/polars-core/src/testing.rs

```diff
@@ -5,7 +5,7 @@ use std::ops::Deref;
 impl Series {
     /// Check if series are equal. Note that `None == None` evaluates to `false`
     pub fn series_equal(&self, other: &Series) -> bool {
-        if self.null_count() > 0 || other.null_count() > 0 {
+        if self.null_count() > 0 || other.null_count() > 0 || self.dtype() != other.dtype() {
```

Comment:
I realize that this is the opposite of what py-polars does, hence the failing tests for python.
```
    def series_equal(
        self, other: "Series", null_equal: bool = False, strict: bool = False
```
@ritchie46 I wonder what we should do here, add it as an argument here too, change the default so that it's True in python too (which would mean making the dtypes explicit there too).

Reply:
Hmm... Maybe we should remove the schema check for now and open a new PR for that refactor?

Reply:
Makes sense, I'll transfer this to another PR.